### ISSUE TITLE
Is there a technical reason sample is not implemented for DiagonalNormal?

### ISSUE THREAD

**Issue by szvsw** [FabricioArendTorres/FlowConductor#4]
Hi there,

Awesome work on this repo.  

I'm brand new to normalizing flows but it seems like they are the exact tool I am looking for.  Essentially, I have an expensive engineering simulation function f:X -> Y where X is [0,1]^19 and Y is [0,1]^24, a surrogate f_hat trained to extremely low MSE on simulation results over a uniform random sample of X, and my goal is to learn p(x | y) so that I can deploy a model to production which, given a query vector y, instantly will produce an arbitrary number of samples from the conditional probability distribution.  Note that the in production, the query vectors are coming from real world data, so while training the normalizing flows, I am adding some noise to the context vector inputs. 

I tweaked the conditional toy 2d build_flow helper to be able to accept my data shapes, and tweaked the training loop to use infinitely generated data from my surrogate, and, much to my amazement, the network seems to be training completely normally and to very low negative log probs - much lower than all of the other methods I have tried!  It is also seems to be behaving as expected, where on my test data, it correctly predicts a very low neg log prob for the true x vector, and as I perturb the true vector x by adding noise to it (deviating away from the true design), the neg log prob increases.  This is all very cool.

However, I was excited to see in the Readme that you could simply sample from p(x | y) once trained; having said that, when I tried, I get a not implemented error due to the following line:

https://github.com/FabricioArendTorres/FlowConductor/blob/df2f366193e9314ba47b81c84f8a3fc213117b04/flowcon/distributions/normal.py#L177

I returned to the original toy conditional 2d problem and confirmed that the same occurs there.  

Just wanted to check to see if there was a specific technical reason why this is not implemented, or if it is just due to dev availability.

**Comment by FabricioArendTorres** [comment]
Hi,

Thanks for your feedback.

There is no technical reason, and it should be easily fixable - I think this was mostly an oversight.
I can see if I can add it in the next few days - else maybe try using `StandardNormal`, it should not affect the behaviour of the flow too much.

Regarding your application:
If you notice the model overfitting on your simulation data, try reducing the flexibility (i.e. size/layers/...) of the conditioning network. 
I assume this was the case, since you add noise to `x`, which has this effect of smoothing f(x|y) around the region of your data - something the embeddng network should (ideally) do on its own.

Some things you can try in general with the densenet from the example:
- Reduce the ``Sin(10)`` to something like `Sin(5)` to limit the capacity of the network.
  This reduces high-frequency information in the activations.
- Fewer layers and smaller sizes of the embedding network(s) - these are the things that may be overly sensitive to input changes. Specifically consider these parameters from the example
```                                  
densenet_depth=3,
densenet_growth=32,
c_embed_hidden_sizes=(128, 128, 10),
m_embed_hidden_sizes=(128, 128)


num_shared_embedding=50
```
